### ISSUE TITLE
bug: input type="time", "date", or "datetime-local" is not aligned in Safari (iOS and macOS)

### ISSUE THREAD

**Issue by cmaas** [ionic-team/ionic-framework#28383]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

v7.x

### Current Behavior

In Safari, the selected date is not aligned properly when using `type="time"`:

```html
<ion-list>
  <ion-item>
    <ion-input label="Set the time" type="time"></ion-input>
  </ion-item>
</ion-list>
```

Looks like this:
![CleanShot 2023-10-19 at 10 06 15@2x](https://github.com/ionic-team/ionic-framework/assets/6094364/72737d8a-1ee5-453a-8b57-6cb0c1d60fa6)

Tested on:
Safari 16.5.1 macOS 13.4.1
iOS Safari 15.7.1
iOS Safari 17.0.3

### Expected Behavior

Correct alignment

### Steps to Reproduce

Load the testing URL in Safari (iOS, macOS)

### Code Reproduction URL

https://flems.io/#0=N4IgzgpgNhDGAuEAmIBcIB0ALeBbKIANCAGYCWMYaA2qAHYCGuEamO+RIsA9nYn6wA87KAD4AOnQAEU4RAZIJ0mbPhl4MUQEleZWFIAqEMPEEB6NRohKVssLABOZAA7wp8AJ7OIAXnEhcbiQAVxh-KTAHWD8QHHhnMFQzM1gkOgwAKzAkaDIANwcMOgh4MzpnXDMAATJdWBTuBwgzJDITM1q6PQ66jGNcTLB-UXN7J1cbFUEoMjoAaykmqBiTD0osCBLwrCaSGLiEpJS0wZyZgqKSsorqzu6eJpSwMB6u2AwAI2C6JBgMWGe4TMk3MGwUSkEHyCHhBnQAtGCcg5JrZBPD4NxuFAPgwHFIeFBGjFnE5cLiPMNJLZqbJ0epNAAJbjMcx0qwoqavOEYrE45FU2RcxEQfnKNG8OE8PgQPj4qAMZ4xeHOBStOgAc0pylRzlEAHVoDxmO5uFIAMr8WAUKQAQWcznN8FxiEK5l1AtR8JmJg5NPFdDh6gguF9NKm8NmzmCbnlH2gMTNJXcG3cZGY4U83hianTIBGXMj0dDqILiBDHs5Xra8BBXKl-BrknMUKQMKbZhESkknEgMAQnSo6AAjAAWVAANjhQ4AHKghwB2EAAX0I9CYLHQg049Zl8FYy9XIEYzFY-2e294Df3S4AukugA

### Ionic Info

Uses the CDN bundle (current version)

**Comment by cmaas** [comment]
If I toggle `display: inline-block` in Safari's webdev tools, it works. However, the same CSS setting doesn't do anything if the input type is`text`. Weird. Video:

https://github.com/ionic-team/ionic-framework/assets/6094364/b3ae251f-0b20-4392-b7ee-93e8d78655c9

**Comment by cmaas** [comment]
This fixes it for now, for whatever reason:

```css
input[type=time].native-input {
    align-self: center;
}
```

**Comment by thetaPC** [comment]
Thank you for submitting the issue! I was able to reproduce the bug on Safari 15, 16, and 17. This is also happening for an input with the type `date` and `datetime-local`. The issue is being caused by the pseudo selector that Safari adds to these types.

**Comment by cmaas** [comment]
[Here's another weird demo.](https://flems.io/#0=N4IgzgpgNhDGAuEAmIBcIB0ALeBbKIANCAGYCWMYaA2qAHYCGuEamO+RIsA9nYn6wA87KAD4AOnQAEU4RAZIJ0mbPhl4MUQEleZWFIAqEMPEEB6NRohKVssLABOZAA7wp8AJ7OIAXnEhcbiQAVxh-KTAHWD8QHHhnMFQzM1gkOgwAKzAkaDIANwcMOgh4MzpnXDMAATJdWBTuBwgzJDITM1q6PQ66jGNcTLB-UXN7J1cbFUEoMjoAaykmqBiTD0osCBLwrCaSGLiEpJS0wZyZgqKSsorqzu6eJpSwMB6u2AwAI2C6JBgMWGe4TMk3MGwUSkEHyCHhBnQAtGCcg5JrZBPD4NxuFAPgwHFIeFBGjFnE5cLiPMNJLZqbJ0epNAAJbjMcx0qwoqavOEYrE45FU2RcxEQfnKNG8OE8PgQPj4qAMZ4xeHOBStOgAc0pylRzlEAHVoDxmO5uFIAMr8WAUKQAQWcznN8FxiEK5l1AtR8JmJg5NPFdDh6gguF9NNpEtmzmCbnlH2gMQAmtxgni1MxwrB5Yr-OiIAAPeBwpzqnAZmC4rTlaMxeAOYIQcKebw1sjpkBSPIMKD1mIABgAHKgACwAVmGrIjVfgodRXKDIY9Yf9gcQC+1YZky8j0aksfj-iTKfc+fgZYVYCVEsQBaLZBLp-bmfkDkrUfgNbrDfbTd8-mvD47Lse38AdhzHEARjnKcZ05eF5xgzc4NXBCpjgqddwYONlgPZM8SQBhEDPbMQFzG9i1LR9yxfKcP3rRsvF-EB8MI9tO27RiACZew4gBmOEAEZ+IEnjxygt8UInAN4MXTcuW9acBUkyVeH4BS6HMKEkBhSRQTwMRJEkThIBgBBOiodBe1QXs4Q4gBOKyQAAX0IegmBYdBBk4KVVNYJyXJARhmFYf5ni8lSZXgXzHIAXUcoA)

I explicitly set the class `ion-text-right` to all three elements. The right-alignment works on macOS (Safari, Firefox, Chrome), but fails on an iPhone with iOS 17 (Safar) as well as a Google Pixel (latest Chrome). I don't know why, but presumably, some pseudo selector stuff interferes here as well, except this time, also in Chrome 😵‍💫

**Comment by mapsandapps** [comment]
Hi, thank you for submitting this issue. This bug appears to have been fixed in Ionic Framework 7.6.0. Please let me know if you're still experiencing issues after updating to the latest version.

| 7.5 | 7.6 |
|---|---|
| ![Screenshot 2024-01-17 at 1 45 51 PM](https://github.com/ionic-team/ionic-framework/assets/14926794/45c32c8c-3a40-452b-889f-04159380b539) | ![Screenshot 2024-01-17 at 1 45 24 PM](https://github.com/ionic-team/ionic-framework/assets/14926794/407a3b48-4c82-4ab6-9aed-3ede708a2198) |
